src/frontend/apps/lti_site/components/App/AppContentLoader/index.tsx

```diff
@@ -104,8 +90,43 @@ const AppContentLoader = () => {
     useCurrentUser.setState({
       currentUser,
     });
+  }, [decodedJwt]);
+
+  return <Content />;
+};
 
+const AppContentLoader = () => {
+  const [isLoaded, setIsLoaded] = useState(false);
+  const queryClient = new QueryClient();
+
+  const defaultResourceContext: ResourceContext = {
+    permissions: {
+      can_access_dashboard: false,
+      can_update: false,
+    },
+    resource_id: '',
+    roles: [],
+  };
+
+  //  load it from the store to prevent having a dependancy and recompute decodedJwt
+  const decodedJwt = useMemo(() => {
+    const jwt = useJwt.getState().jwt;
+    if (jwt) {
+      setIsLoaded(true);
+      return decodeJwt(jwt);
+    }
     setIsLoaded(true);
+  }, []);
+
+  const intl = useMemo(() => {
+    return createIntl(decodedJwt ? decodedJwt.locale : 'en');
+  }, [decodedJwt]);
+
+  const resourceContext: ResourceContext = useMemo(() => {
+    if (decodedJwt) {
+      return { ...decodedJwt };
+    }
+    return defaultResourceContext;
```

Comment:
'defaultResourceContext' should be added in the dependency.

Reply:
Event if it is a constant that I set ? It's not supposed to change over time